app/models/manageiq/providers/vmware/infra_manager/event_catcher.rb

```diff
@@ -1,3 +1,38 @@
 class ManageIQ::Providers::Vmware::InfraManager::EventCatcher < ManageIQ::Providers::BaseManager::EventCatcher
   require_nested :Runner
+
+  def self.unit_file
+    <<~UNIT_FILE
+      [Unit]
+      PartOf=#{target_file_name}
+      [Install]
+      WantedBy=#{target_file_name}
+      [Service]
+      WorkingDirectory=#{working_directory}
+      ExecStart=/bin/bash -lc '#{exec_start}'
+      Restart=always
+      Slice=#{slice_name}
+    UNIT_FILE
+  end
+
+  def self.working_directory
+    ManageIQ::Providers::Vmware::Engine.root.join("workers/event_catcher")
+  end
+
+  def self.exec_start
+    "exec ~/.gem/bin/bundle exec ruby event_catcher.rb"
```

Comment:
Not sure I like this approach, but for now it's fine.

Reply:
yeah the `~/.gem/bin/bundle` part is just because I was struggling with getting `$PATH` working when running as a non-root systemd unit and won't be needed